syntax = "proto3";

package Appearance;

enum OperateType {
  SEND_INFO = 0;
  ACTIVE_AVATAR = 1;
  USE_AVATAR = 2;
  ACTIVE_AVATAR_FRAME = 3;
  USE_AVATAR_FRAME = 4;
}

message CS_AppearanceRequest {
  OperateType operate_type = 1;
  uint32 index = 2;
  int32 parameter = 3;
}

message SC_AppearanceResponse {
  message Avatar {
    uint32 index = 1;
    bool bActivated = 2;
    uint64 expired = 3;
  }

  message AvatarFrame {
    uint32 index = 1;
    bool bActivated = 2;
    uint64 expired = 3;
  }

  uint32 current_avatar = 1;
  uint32 current_avatar_frame = 2;

  repeated Avatar avatar = 3;
  repeated AvatarFrame avatar_frame = 4;
}